src/listmode_buildblock/CListModeDataGESigna.cxx

```diff
@@ -0,0 +1,131 @@
+/*
+    Copyright (C) 2013 University College London
+*/
+/*!
+  \file
+  \ingroup listmode  
+  \brief Implementation of class stir::CListModeDataGESigna
+    
+  \author Kris Thielemans
+*/
+
+
+#include "stir/listmode/CListModeDataGESigna.h"
+#include "stir/Succeeded.h"
+#include "stir/ExamInfo.h"
+#include "stir/info.h"
+#include "stir/IO/HDF5Wrapper.h"
+#include <boost/format.hpp>
+#include <iostream>
+#include <fstream>
+
+START_NAMESPACE_STIR
+
+CListModeDataGESigna::
+CListModeDataGESigna(const std::string& listmode_filename)
+  : listmode_filename(listmode_filename)    
+{
+  if (open_lm_file() == Succeeded::no)
+    error(boost::format("CListModeDataGESigna: error opening the first listmode file for filename %s") %
+	  listmode_filename);
+ printf( "\n Success in opening the listmode\n" );
+}
+
+std::string
+CListModeDataGESigna::
+get_name() const
+{
+  return listmode_filename;
+}
+
+std::time_t 
+CListModeDataGESigna::
+get_scan_start_time_in_secs_since_1970() const
+{
+  return std::time_t(-1); // TODO
+}
+
+
+shared_ptr <CListRecord> 
+CListModeDataGESigna::
+get_empty_record_sptr() const
+{
+  shared_ptr<CListRecord> sptr(new CListRecordT);
+  return sptr;
+}
+
+Succeeded
+CListModeDataGESigna::
+open_lm_file()
+{
+  info(boost::format("CListModeDataGESigna: opening file %1%") % listmode_filename);
+#if 0
+  shared_ptr<std::istream> stream_ptr(new std::fstream(listmode_filename.c_str(), std::ios::in | std::ios::binary));
+  if (!(*stream_ptr))
+    {
+      return Succeeded::no;
+    }
+  stream_ptr->seekg(12492704); // TODO get offset from RDF. // I got it from the listmode OtB 1/09/16 5872
+  current_lm_data_ptr.reset(
+                            new InputStreamWithRecords<CListRecordT, bool>(stream_ptr, 
+                                                                           4, 16, 
+                                                                           ByteOrder::little_endian != ByteOrder::get_native_order()));
+#else
+  if(!HDF5Wrapper::check_GE_signature(listmode_filename))
+  {
+     //! \todo N.E:Write a msg
+     return Succeeded::no;
+  }
+
+//  input_sptr.reset( new HDF5Wrapper(listmode_filename));
+
+
+#endif
+
+  //! \todo N.E: Probably can do without the HDF5Wrapper here.
+  HDF5Wrapper inputFile(listmode_filename);
+  CListModeData::scanner_sptr = inputFile.get_scanner_sptr();
+  
+  //! \todo N.E: Remove hard-coded sizes;
+  current_lm_data_ptr.
+  reset(
+        new InputStreamWithRecordsFromHDF5<CListRecordT>(listmode_filename,
```

Comment:
are we sure about this? I thought they're always 6 bytes? if you don't know, put a todo comment

Reply:
Palak did confirm this is ok